### PR TITLE
Enable CSRF for SIC Code Web

### PR DESCRIPTION
**Pull request by nawazish587**
__Short description outlining key changes/additions__
Enable CSRF for SIC Code Web

__JIRA Ticket Number__
Resolves: SEC-84, 85 and 86

### Type of change

* [ ] Bug fix
* [X] New feature
* [ ] Breaking change

### Pull request checklist

* [X] I have added unit tests for new code that I have added
* [ ] I have added/updated functional tests where appropriate
* [X] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/javascript_node.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,8 +28,12 @@
   },
   "homepage": "https://github.com/companieshouse/sic-code-web#readme",
   "dependencies": {
+    "@companieshouse/node-session-handler": "^5.0.1",
     "@companieshouse/structured-logging-node": "^1.0.4",
+    "@companieshouse/web-security-node": "^4.1.1",
     "axios": "^1.6.7",
+    "chai": "^5.1.1",
```
Comment (JoelClemence) [inline package.json:35]: chai is a testing library so should be in `devDependencies` - and appears to be there already so you can delete this line

**Review by JoelClemence — COMMENTED** [review]
Can you review the dependencies you have added to confirm they are necessary and remove chai from dependencies as it will be installed as part of the apps dependencies

File: package.json [hunk 2]
```diff
@@ -38,17 +42,27 @@
...
     "@types/jest": "^27.4.1",
     "@types/node": "^18.11.18",
     "@types/nunjucks": "^3.2.1",
+    "@types/sinon": "~7.5.2",
+    "@types/sinon-chai": "~3.2.12",
     "@types/supertest": "^2.0.11",
     "@typescript-eslint/eslint-plugin": "^5.16.0",
     "@typescript-eslint/parser": "^5.16.0",
+    "chai": "~4.2.0",
+    "chai-as-promised": "~7.1.1",
```
Comment (JoelClemence) [inline package.json:58]: How are these being used? the tests are using Jest so do you really need to add chai/sinon ? if not I'd suggest removing them.

File: package.json (hunk 1 above)
Comment (nawazish587) [reply package.json]: thanks @JoelClemence I have removed chai dependencies. Unit tests, lint and code are running fine.

File: package.json (hunk 2 above)
Comment (nawazish587) [reply package.json]: thanks @JoelClemence I have removed chai dependencies.

File: src/middleware/createCsrfErrorMiddleware.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { NextFunction, Request, Response } from "express";
+import { CsrfError } from "@companieshouse/web-security-node";
+
+const csrfErrorTemplateName = "403";
+
+const csrfErrorHandler = (
+  err: CsrfError | Error,
```
Comment (pshdev0) [inline src/middleware/createCsrfErrorMiddleware.ts:7]: should we include `| null | undefined` here? I know TS won't allow it, but once transpiled into JS, could a `null` or `undefined` could potentially come in? Same for other middleware I guess...

File: package.json [hunk 4]
```diff
@@ -38,17 +41,20 @@
...
     "@types/nunjucks": "^3.2.1",
+    "@types/sinon": "~7.5.2",
     "@types/supertest": "^2.0.11",
     "@typescript-eslint/eslint-plugin": "^5.16.0",
     "@typescript-eslint/parser": "^5.16.0",
     "copyfiles": "^2.4.1",
     "eslint": "^7.32.0",
     "jest": "^27.5.1",
     "nodemon": "^3.0.1",
+    "sinon": "~9.0.1",
```
Comment (JoelClemence) [inline package.json:57]: Do you need sinon? I don't see it being used in the PR

File: package.json [hunk 5]
```diff
@@ -38,17 +41,20 @@
     "uuid": "^8.3.2"
   },
   "devDependencies": {
+    "@types/cookie-parser": "~1.4.6",
     "@types/express": "^4.17.13",
     "@types/jest": "^27.4.1",
     "@types/node": "^18.11.18",
     "@types/nunjucks": "^3.2.1",
+    "@types/sinon": "~7.5.2",
```
Comment (JoelClemence) [inline package.json:49]: ditto

**Review by JoelClemence — COMMENTED** [review]
There's a dependency on Sinon and I am not sure it is necessary as Jest is used for mocking.

File: package.json (hunk 4 above)
Comment (nawazish587) [reply package.json]: thanks @JoelClemence  I have modified it.

File: package.json (hunk 5 above)
Comment (nawazish587) [reply package.json]: thanks @JoelClemence I have modified it.

**Review by JoelClemence — APPROVED** [review]

**Pull request merged**